### PR TITLE
STFE-52: add classroom cards

### PR DESCRIPTION
**Pull request by mohammed-sharraf-sazim**
## Description of change:
- Add teacher and student classrooms dashboard

## Associated Trello Link(s):
- https://trello.com/c/dE7O4CHj/52-classroom-card-components-fe

## Related Pull Request(s):
- https://github.com/mohammed-sharraf-sazim/SuperTeacher-FE/pull/17

## Screenshots
- 
Screenshot 2024-09-12 at 12 40 58 PM
- 
Screenshot 2024-09-12 at 12 41 19 PM

- 
Screenshot 2024-09-12 at 12 42 02 PM

### CODE REVIEW

File: modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx [hunk 1]
```diff
@@ -1,20 +1,42 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 
+import { convertTimeToDate } from "@/modules/AddClassroom/components/ClassroomForm.helper";
 import { ClassroomModal } from "@/modules/AddClassroom/components/ClassroomModal";
+import ClassroomCard from "@/modules/ClassroomCard/ClassroomCard";
```
Comment (mahir-mohsin-sazim) [inline modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx:5]: use Default export

File: modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
 import { useAppSelector } from "@/shared/redux/hooks";
 import { authenticatedUserSelector } from "@/shared/redux/reducers/user.reducer";
 import { EUserRole } from "@/shared/redux/rtk-apis/auth/auth.types";
+import { useGetClassroomsQuery } from "@/shared/redux/rtk-apis/classrooms/classrooms.api";
+import { Classroom } from "@/shared/redux/rtk-apis/classrooms/classrooms.types";
 
 import Navbar from "../../components/Navbar/Navbar";
```
Comment (mahir-mohsin-sazim) [inline modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx:12]: use default export

File: modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx [hunk 3]
```diff
@@ -1,20 +1,42 @@
...
     const user = useAppSelector(authenticatedUserSelector);
+    const {data} = useGetClassroomsQuery();
+    const [classrooms, setClassrooms] = useState<Classroom[]>([]);
+
+    useEffect(() => {
+        if (data) {
+            setClassrooms(data || []);
+        }
+    }, [data]);
+    console.log(data);
```
Comment (mahir-mohsin-sazim) [inline modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx:24]: remove this

**Review by mahir-mohsin-sazim — CHANGES_REQUESTED** [review]
Please if for mobile screens, all your cards are center aligned or not. From the images you have attached, I do not think they are properly aligned

File: modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx (hunk 2 above)
Comment (mohammed-sharraf-sazim) [reply modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx]: resolved

File: modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx (hunk 3 above)
Comment (mohammed-sharraf-sazim) [reply modules/UserDashboard/containers/ClassroomContainer/ClassroomContainer.tsx]: removed

**Review by mahir-mohsin-sazim — APPROVED** [review]

File: modules/ClassroomCard/ClassroomCard.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
+import React from "react";
```
Comment (nur-zaman-sazim) [inline modules/ClassroomCard/ClassroomCard.tsx:1]: nit: we don't need this line

File: modules/UserDashboard/components/Navbar/Navbar.tsx [hunk 5]
```diff
@@ -1,22 +1,36 @@
-import React from "react";
+import React, { useState } from "react";
```
Comment (nur-zaman-sazim) [inline modules/UserDashboard/components/Navbar/Navbar.tsx:1]: nit: same comment as before

**Review by nur-zaman-sazim — APPROVED** [review]
Left some nitpicks

**Pull request merged**